core/templates/components/question-directives/question-player/services/question-player-state.service.spec.ts

```diff
@@ -266,4 +271,13 @@ describe('Question player state service', function() {
       expect(stateData[questionId].linkedSkillIds).toEqual(
         ['skill_id1', 'skill_id2']);
     });
+
+  it('should emit the question session completed event correctly', () => {
+    testSubscriptions = new Subscription();
+    testSubscriptions.add(
+      qpservice.onQuestionSessionCompleted.subscribe(
+        questionSessionCompletedSpy));
+    qpservice.onQuestionSessionCompleted.emit();
+    expect(questionSessionCompletedSpy).toHaveBeenCalled();
+  });
```

Comment:
I'm not sure what purpose this test is serving. If I understand correctly, you are emitting and subscribing in the test itself. Is this just a test for the getter function? If so, is it not straightforward to just check the getter function returns an EventEmitter?

Reply:
Done